docs/copied-from-beats/outputconfig.asciidoc

```diff
@@ -739,21 +739,41 @@ use in Logstash for indexing and filtering:
 for more about the `@metadata` field.
 <2> The default is {beat_default_index_prefix}. To change this value, set the
 <<logstash-index,`index`>> option in the {beatname_uc} config file.
-<3> The beats current version.
-<4> The value of `type` is currently hardcoded to `doc`. It was used by previous
-Logstash configs to set the type of the document in Elasticsearch.
-
+<3> The current version of {beatname_uc}.
+endif::[]
 
-WARNING: The `@metadata.type` field, added by the Logstash output, is
-deprecated, hardcoded to `doc`, and will be removed in {beatname_uc} 7.0.
+ifdef::apm-server[]
+["source","json",subs="attributes"]
+------------------------------------------------------------------------------
+{
+    ...
+    "@metadata": { <1>
+      "beat": "{beat_default_index_prefix}", <2>
+      "pipeline":"apm", <3>
```

Comment:
I thought automatic pipeline registration requires `output.elasticsearch` to be `true`? Why am I seeing this here?

Reply:
The registration of the pipeline does require ES output, as it registers the pipeline on startup/setup. Applying pipelines on events during ingestion is also possible for logstash output.